src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -2,20 +2,62 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
 namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
         private static bool GetInvariantSwitchValue() =>
             GetSwitchValue("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
 
+        private static bool TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> icuSuffix)
+        {
+            icuSuffix = default;
+
+            if (!TryGetStringValue("System.Globalization.AppLocalIcu", "DOTNET_SYSTEM_GLOBALIZATION_APPLOCALICU", out string? value))
+            {
+                version = default;
+                return false;
+            }
+
+            // Custom built ICU can have a suffix on the name, i.e: libicuucmyapp.so.67.1
+            // So users would set the runtime switch as: myapp:67.1
+
+            int indexOfSeparator = value.IndexOf(':', StringComparison.Ordinal);
+            if (indexOfSeparator != -1)
+            {
+                ReadOnlySpan<char> valueAsSpan = value.AsSpan();
+                icuSuffix = valueAsSpan.Slice(0, indexOfSeparator);
+
+                if (icuSuffix.Length > 20)
```

Comment:
>20 [](start = 39, length = 2)

I think 20 is a small number for suffix. I would say make it like 35 or so.

Reply:
Ok, sounds good.